mountpoint-s3/src/inode/expirable_set.rs

```diff
@@ -0,0 +1,90 @@
+use std::time::Duration;
+
+use linked_hash_map::LinkedHashMap;
+use std::time::Instant;
+
+use super::{expiry::Expiry, InodeNo};
+
+use crate::sync::RwLock;
+
+/// A bounded set of (parent_ino, child_name) entries that expire after a fixed time.
+#[derive(Debug)]
+pub struct ExpirableSet {
+    map: RwLock<LinkedHashMap<Key, Expiry>>,
+    max_size: usize,
+    validity: Duration,
+}
+
+#[derive(Debug, Hash, PartialEq, Eq)]
+struct Key {
+    parent_ino: InodeNo,
+    child_name: String,
+}
+
+impl ExpirableSet {
+    pub fn new(max_size: usize, validity: Duration) -> Self {
+        Self {
+            map: RwLock::new(Default::default()),
+            max_size,
+            validity,
+        }
+    }
+
+    pub fn contains(&self, parent_ino: InodeNo, child_name: &str) -> bool {
+        let key = Key {
+            parent_ino,
+            child_name: child_name.to_owned(),
```

Comment:
This copy annoys me quite a bit. Probably not the most critical optimization, but is there a simple way to avoid it?